exercises/ayomide_bamidele/ex02_oo_basics/src/Square.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Square.h"
+#include <iostream>
+
+Square::Square(double x) : length(x) {}
+
+double Square::area() const { return length * length; }
+
+double Square::perimeter() const { return 4 * length; }
```

Comment:
Sides is defined to be 4. Why not use `sides * length`

Reply:
Well, that's true. I guess I like preferred it this way because it's slicker mathematically. I only have the sides variable because it is required to solve the question.

Reply:
Is using `sides * length` better ?

Reply:
> I guess I like it this way because it's slicker mathematically: You only use one variable.

You might think so, but I would argue that this is not true. Compilers are very smart and good at optimising your code. With optimisation flags turned on a good compiler is going to see that variable and optimise it away. 

Don't believe me? Here's your [code vs. mine](https://godbolt.org/g/E5iMem) in the extremely useful [compiler explorer](https://godbolt.org/). The assembly on the right shows that clang 6.0.0 outputs the same code regardless of whether we use a literal or a const member variable.

So I would argue the we should use `sides * length` because it's more descriptive (even thought yes I can guess 4 means the number of sides).

Even if this was not true and using the member variable was slightly slower I would still argue that you use the variable over the literal. In general prefer to write good readable code over "fast" code. Then *measure* the performance and think about how you can improve speed *if it matters*. If code isn't in a critical high performance area then it doesn't need to be the fastest possible code; it is more important that it is clear, readable and maintainable.

Reply:
Alright, I understand this.